### PR TITLE
Make source argument to our Dagger module optional

### PR DESCRIPTION
**Pull request by shykes**
*This PR depends on #8241. Please review that first*.

This uses the new "context directory" feature in Dagger 0.12.6

### CODE REVIEW

**Review by vito — APPROVED** [review]
cute-baby-cheering-enkl65ixwvswlna1

File: modules/compatcheck/main.go
Comment (jedevc) [inline modules/compatcheck/main.go:62]: Nifty :tada:

File: .github/main.go
Comment (jedevc) [inline .github/main.go:77]: Some other places can be updated too:
- `action.yml` (which is still used by some of the workflows)
- `.dagger/mage/util` (though that code's day's are very very numbered)
- `README.md` (in `.dagger` and `sdk/python`)

**Review by jedevc — APPROVED** [review]
One minor note, otherwise LGTM once we get #8241 in :tada:

File: .github/main.go
Comment (shykes) [reply .github/main.go:77]: on it

**Comment by shykes** [comment]
@helderco I also made the change for the python sdk dev module as well

File: .github/main.go
Comment (shykes) [reply .github/main.go:77]: done, thanks!

**Comment by shykes** [comment]
It's too much work to maintain this + #8241 until it gets merged, so I'm combining them. Closing this, and moving the action to #8241

**Comment by jedevc** [comment]
:cry: wait why wasn't this just merged? It had an approval.

**Pull request closed without merging**
